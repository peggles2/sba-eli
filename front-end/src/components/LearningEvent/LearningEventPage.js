import React, { Component } from "react";
import "./LearningEventPageContent.scss";

import { completeLearningEvent } from "../../actions/learningEventActions";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";

export class LearningEventPage extends Component {
  componentDidMount() {
    this.completeEvent();
  }

  componentDidUpdate(prevProps) {
    if (this.props.event !== prevProps.event) {
      this.completeEvent();
    }
  }

  completeEvent() {
    const { isUserLoggedIn, event } = this.props;
    if (isUserLoggedIn && !event.completion_requirement.completed) {
      const { id: path_id, topicId: objective_id } = this.props.match.params;

      this.props.dispatch(
        completeLearningEvent(path_id, objective_id, event.id)
      );
    }
  }

  render() {
    const __html = this.props.event.eventContent.body;

    return (
      <div
        className="learning-event-body"
        dangerouslySetInnerHTML={{
          __html: __html
        }}
      />
    );
  }
}

const mapStateToProps = store => {
  return { isUserLoggedIn: store.login.isUserLoggedIn };
};

export default withRouter(connect(mapStateToProps)(LearningEventPage));
