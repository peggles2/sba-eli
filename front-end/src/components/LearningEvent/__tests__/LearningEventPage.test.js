import React from "react";
import { LearningEventPage } from "../LearningEventPage";
import { shallow } from "enzyme";

describe("LearningEventPage", () => {
  it("should render correctly", () => {
    const event = { eventContent: { body: "<div></div>" } };
    const wrapper = shallow(<LearningEventPage event={event} />);

    expect(wrapper.find("div").exists()).toBe(true);
  });
});
