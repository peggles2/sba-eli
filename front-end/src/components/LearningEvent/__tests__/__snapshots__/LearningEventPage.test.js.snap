// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningEventPage should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningEventPage
    event={
      Object {
        "eventContent": Object {
          "body": "<div></div>",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div></div>",
          }
        }
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "dangerouslySetInnerHTML": Object {
          "__html": "<div></div>",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<div></div>",
            }
          }
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "<div></div>",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
